finix/model/finix_list.py

```diff
@@ -0,0 +1,171 @@
+
+
+# custom exception raised when list_next() fails
+class PaginationException(Exception):
+    def __init__(self):
+        self.message = 'list_next() fails: either pagination information is missing or there is no more resource to fetch!'
+    
+    def __str__(self):
+        return self.message
+
+
+# finix object implementation
+class FinixObject:
+    def __init__(self,some_dict):
+        self.__dict__.update(some_dict)
+
+    def __str__(self):
+        return  str(self.__class__) + '\n'+ '\n'.join(('{} : {}'.format(item, self.__dict__[item]) for item in self.__dict__))
+
+
+# finix error implementation
+class FinixError:
+    def __init__(self, some_dict):
+        self.__dict__.update(some_dict)
+
+        if 'authorization' in some_dict:
+            self.authorization = some_dict['authorization']
+        else:
+            self.authorization = None
+        
+        if 'code' in some_dict:
+            self.code = some_dict['code']
+        else:
+            self.code = None
+
+        if 'failure_code' in some_dict:
+            self.failure_code = some_dict['failure_code']
+        else:
+            self.failure_code = None
+
+        if 'failure_message' in some_dict:
+            self.failure_message = some_dict['failure_message']
+        else:
+            self.failure_message = None
+
+        if 'logref' in some_dict:
+            self.logref = some_dict['logref']
+        else:
+            self.logref = None
+
+        if 'message' in some_dict:
+            self.message = some_dict['message']
+        else:
+            self.message = None
+
+        if 'transfer' in some_dict:
+            self.transfer = some_dict['transfer']
+        else:
+            self.transfer = None
+
+        if 'field' in some_dict:
+            self.field = some_dict['field']
+        else:
+            self.field = None
+
+    def __str__(self):
+        return  str(self.__class__) + '\n'+ '\n'.join(('{} : {}'.format(item, self.__dict__[item]) for item in self.__dict__))
+
+# error list implementation
+class ErrorList:
+    errors: List[FinixError]
+
+    def __init__(self, some_dict):
+        self.errors = []
+        if isinstance(some_dict, dict):
+            if '_embedded' in some_dict:
+                if 'errors' in some_dict['_embedded']:
+                    if isinstance(some_dict['_embedded']['errors'], list):
+                        for item in some_dict['_embedded']['errors']:
+                            self.errors.append(FinixError(item))
+        if self.errors == []:
+            self.no_error_body = True
+        else:
+            self.no_error_body = False
+                     
+    def __len__(self):
+        return len(self.errors)
+    
+    def __getitem__(self,index):
+        return self.errors[index]
+
+    def __iter__(self):
+        return iter(self.errors)
+
+    def __str__(self):
+        return "This is an ErrorList object, please customize '__str__' by yourself!"
```

Comment:
nitpick: Do we expect this method to be overridden? If not, why can't we implement `__str__`? Same with FinixList above.

Reply:
I did this since I can't think of an elegant yet simple way to print those Lists

I guess users won't directly print a list very often; if they do, I expect them to override this method

Reply:
I'd just remove __str__ then. We can write a proper __str__ method later if we find we need one.